spyder/widgets/ipythonconsole/debugging.py

```diff
@@ -52,10 +54,14 @@ def refresh_from_pdb(self, pdb_state):
             self.sig_pdb_step.emit(fname, lineno)
 
         if 'namespace_view' in pdb_state:
-            self.sig_namespace_view.emit(pdb_state['namespace_view'])
+            if pdb_state['namespace_view']:
```

Comment:
Why did you need to add this `if` clause here? I think we need to emit whatever namespace contents comes from the kernel.